### PR TITLE
Validation added, fingerprintfield updated

### PR DESCRIPTION
**Pull request by vitiwari**
@gokumar 
Added validation for fingerPrintField rule & app_id.

### CODE REVIEW

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java [hunk 1]
```diff
@@ -34,11 +34,18 @@ public boolean validate(Template template) throws ValidationException {
...
-        // validate payload configuration
         for (String fpField : payload.getFingerprintFields()) {
-            if (fpField != null && fpField.startsWith("@") && !fieldItemMap.containsKey(fpField)) {
-                throw new ValidationException(
-                        StringUtil.format(Constants.PAYLOAD_PLACEHOLDER_DEFINITION_MISSING, new Object[]{fpField}));
+            if (fpField != null) {
+                if (fpField.startsWith("@")) {
+                    String field = fpField.substring(1);
+                    if (!Constants.FINGERPRINT_EVENT_FIELDS.contains(field)) {
+                        throw new ValidationException(StringUtil.format(Constants.EVENT_FIELD_MISSING, new Object[]{field, Constants.FINGERPRINT_EVENT_FIELDS}));
```
Comment (gokumar) [inline src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java:42]: Should `Constants.EVENT_FIELD_MISSING` change to something like `Constants.INVALID_FINGERPRINT_FIELD_FOUND`???

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java [hunk 2]
```diff
@@ -34,11 +34,18 @@ public boolean validate(Template template) throws ValidationException {
...
+                    }
+                } else {
+                    if (!payload.getProperties().containsKey(fpField)) {
+                        throw new ValidationException(StringUtil.format(Constants.EVENT_PROPERTY_FIELD_MISSING, new Object[]{fpField}));
```
Comment (gokumar) [inline src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java:46]: `Constants.EVENT_PROPERTY_FIELD_MISSING` also needs rewording

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java [hunk 3]
```diff
@@ -100,5 +111,12 @@ public boolean validate(Template template) throws ValidationException {
         }
         return true;
     }
-
+    private boolean isValidAppId(String inputString) {
+        for (char c : inputString.toCharArray()) {
+            if (Constants.SPECIAL_CHARACTOR.indexOf(c, 0) >= 0) {
```
Comment (gokumar) [inline src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java:112]: `Constants.SPECIAL_CHARACTOR` -- typo... should be `CHARACTER`

**Review by gokumar — CHANGES_REQUESTED** [review]

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java (hunk 1 above)
Comment (vitiwari) [reply src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java:42]: Constants.FINGERPRINT_EVENT_FIELDS contains a list of allowed event fields.

File: src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java (hunk 2 above)
Comment (vitiwari) [reply src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java:46]: May be we can reword it, it just suggests that the field does not exists in properties field of eventDefinition.

**Pull request merged**